lib/kafo_parsers/puppet_strings_module_parser.rb

```diff
@@ -55,8 +55,14 @@ def initialize(file)
 
     # AIO and system default puppet bins are tested for existence, fallback to just `puppet` otherwise
     def self.puppet_bin
+      bin_dir = if File.exist?('/opt/theforeman/tfm/root/usr/bin')
```

Comment:
My first reaction on this is :-1:. First of all, PATH already has a built in lookup mechanism so there's really no point in replicating it. You generally can't just use files from `/opt/theforeman/tfm/root/usr/bin` without activating the SCL. That also adds the directory to `$PATH` which removes the need for this patch altogether. It should automatically be picked up. Kafo parsers is also a generic gem that's usable outside of the Foreman ecosystem and we shouldn't be hardcoding things to it.

Reply:
I can't tell if you are :-1: the implementation or the *idea*. I have no doubts this ain tthe best implementation for figuring out which one to use. If its the *idea* itself, please comment on the [RFC](https://community.theforeman.org/t/rfc-drop-reliance-on-puppet-agent-aio-for-foreman-installer/16987) as this is just intended to show a working implementation.

You can just use files at that path if those files use `tfm-ruby` in the shebang which is what the RPM packaging PR is doing.

Reply:
At least the implementation. For the idea I posted https://community.theforeman.org/t/rfc-drop-reliance-on-puppet-agent-aio-for-foreman-installer/16987/8